src/auth/src/token.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Represents an auth token.
+#[derive(Debug)]
+pub struct Token {
+    /// The actual token string.  This is the value used in Authorization header.
```

Comment:
Access/ID tokens always use "Authorization" header right, irrespective of platform? (even though ID token is not really authz)

Reply:
I think gRPC uses something else for per-call credentials.

Reply:
I see. that means we would still need `get_token()` so that grpc lib can add whatever header name it wants.
But let us just keep `get_headers()` for now as being discussed in the other comment. We can revisit adding `get_token()` when needed.